### PR TITLE
Make tensorflow requirements optional for building docs

### LINKED ISSUE #416: Master build failing due to tensorflow-quantum incompatibility
Version `0.4.0` works on `py38` but not Windows, and version `0.3.1` works on Windows but not `py38`. I have an idea for a fix, by essentially making the `tensorflow` and `tensorflow-quantum` requirements optional for now.

### PR DESCRIPTION
**Pull request by karalekas**
Description
-----------

Fixes #416.

Checklist
-----------
Please make sure you have finished the following tasks before requesting a review of the pull request (PR). For more information, please refer to the [Contributor's Guide](../blob/master/CONTRIBUTING.md):

- [x] The [changelog](https://github.com/unitaryfund/mitiq/blob/master/CHANGELOG.md) is updated, including author and PR number (@username, gh-xxx).

### CODE REVIEW

**Review by rmlarose — COMMENTED** [review]
What about `dev_requirements.txt` or `doc_requirements.txt` instead of `optional-requirements.txt`? Optional requirements is a bit of an oxymoron.

Alternatively, maybe just pip install TF and TFQ directly in ci.yml since there are only two? This might be desirable if we expect the TFQ compatibility issues to be resolved soon.

File: docs/CONTRIBUTING_DOCS.md [hunk 1]
```diff
@@ -26,9 +26,20 @@ pip install -U sphinx m2r sphinxcontrib-bibtex pybtex sphinx-copybutton sphinx-a
...
+which are used in `guide/guide-executors.rst`. They can be installed via:
+
+```bash
+pip install -r docs/optional-requirements.txt
+```
+
+They are currently optional because `tensorflow-quantum` has incompatibility issues.
+Version `0.4.0` works on `py38` but not Windows, and version `0.3.1` works on Windows
+but not `py38`. As such, the optional requirements as they stand cannot be installed
+on Windows.
```
Comment (rmlarose) [inline docs/CONTRIBUTING_DOCS.md:41]: Should be "As such, they are optional requirements..." I think?

**Comment by karalekas** [comment]
@rmlarose I'm happy to remove the oxymoron :) but I think it's worth keeping them in a file and in the instructions somewhere (rather than just the CI YAML) so that people can run locally what the build runs if they so choose. I can open a follow-up issue to undo this hoop-jumping once TFQ figures out Windows

**Comment by rmlarose** [comment]
> @rmlarose I'm happy to remove the oxymoron :) but I think it's worth keeping them in a file and in the instructions somewhere (rather than just the CI YAML) so that people can run locally what the build runs if they so choose. I can open a follow-up issue to undo this hoop-jumping once TFQ figures out Windows

Sounds good to me.

File: docs/CONTRIBUTING_DOCS.md [hunk 2]
```diff
@@ -26,9 +26,20 @@ pip install -U sphinx m2r sphinxcontrib-bibtex pybtex sphinx-copybutton sphinx-a
 `sphinx-copybutton` allows to easily copy-paste code snippets from examples.
 `sphinx-autodoc-typehints` allows to control how annotations are displayed in the API-doc part of the documentation, integrating with `sphinx-autodoc` and `sphinx-napoleon`.
 
-
 You can check that Sphinx is installed with `sphinx-build --version`.
 
+In addition, there are two requirements, `tensorflow` and `tensorflow-quantum`,
+which are used in solely `guide/guide-executors.rst`. They can be installed via:
```
Comment (rmlarose) [inline docs/CONTRIBUTING_DOCS.md:32]:
```suggestion
which are used solely in `guide/guide-executors.rst`. They can be installed via:
```

**Review by rmlarose — APPROVED** [review]
LGTM.

**Pull request merged**